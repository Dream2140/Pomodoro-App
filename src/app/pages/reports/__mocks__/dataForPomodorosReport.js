import { Helpers } from '../../../helpers/helpers';

export const dataForPomodorosReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 7,
    high: 1,
    low: 0,
    middle: 0,
    urgent: 3,
  },
];
