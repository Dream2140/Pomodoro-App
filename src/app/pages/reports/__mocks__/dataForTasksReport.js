import { Helpers } from '../../../helpers/helpers';

export const dataForTasksReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 1,
    high: 0,
    low: 0,
    middle: 0,
    urgent: 1,
  },
];
