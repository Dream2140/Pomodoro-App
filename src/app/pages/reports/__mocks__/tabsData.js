export const tabsData = {
    leftTabs: false,
    typeTabs: 'reports__top-tabs',
    tabPosition: 'tab__item--centered',
    tabData: [

      {
        period: true,
        val: 'day',
        data: 'Day',
        dataTab: 'data-reports-period="day"'
      },
      {
        period: true,
        val: 'week',
        data: 'Week',
        dataTab: 'data-reports-period="week"'
      },
      {
        period: true,
        val: 'month',
        data: 'Month',
        dataTab: 'data-reports-period="month"'
      }
    ]
};
