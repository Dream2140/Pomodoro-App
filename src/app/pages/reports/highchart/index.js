import './highchart.less';
import Highcharts from 'highcharts';
import Exporting from 'highcharts/modules/exporting';
import { chartOptions } from './chartOptions';
import { chartThemes } from './chartThemes';
import { HighchartController } from './scripts/highchartController';
import { HighchartModel } from './scripts/highchartModel';
import { Observer } from '../../../helpers/observer';
import { HighchartView } from './scripts/highchartView';
Exporting(Highcharts);

const highchartModel = new HighchartModel(
  chartThemes,
  chartOptions,
  Observer,
  Highcharts
);
const highchartView = new HighchartView(highchartModel, Highcharts);

/* eslint-disable no-unused-vars */
const highchartController = new HighchartController(
  highchartModel,
  highchartView
);
/* eslint-disable no-unused-vars */
