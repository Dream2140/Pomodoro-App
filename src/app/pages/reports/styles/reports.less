@import ( reference ) '~@less-helpers-module/_mixins.less'; 
@import ( reference ) '~@less-helpers-module/_variables.less';

.reports {

    &__top-tabs {
        margin-bottom: 213/@rem;
    }

    &__content {
        .centered-flex-element();
        margin-bottom: 191/@rem;
    }

    &__link-left {
        .arrows();
        left: 0;
    }

    &__graph {
        position: relative;
        max-width: 100%;
        max-height: 100%;
        .tooltip();

        .reports-table {
            width: 100%;
            color: @white-color;
            text-align: center;
        }
    }

    &__temp-img {
        margin: 0 auto;
        width: 100%;
        height: auto;
    }

}

.reports-tablet();
.reports-desktop();