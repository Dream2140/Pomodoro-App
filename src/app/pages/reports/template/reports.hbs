{{> header}}
<main class="content reports-page">

       <section class="reports">
        <div class="wrapper">
            <div class="reports__container">
                {{> tab topTabs }}
                <div class="reports__content">
                    <a href="/task-list" class="reports__link-left icon icon-arrow-left" title="Go to Global List"></a>   
                    <div class="reports__graph" id="reportsGraph">
                        
                    </div>
                </div>
                {{> tab bottomTabs }}
            </div>
        </div>
       </section>


    </main>