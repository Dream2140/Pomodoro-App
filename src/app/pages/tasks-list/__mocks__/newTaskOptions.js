export const newTaskOptions = {
  categoryId: 'other',
  completeDate: null,
  completedCount: [],

  deadline: 'April 5, 2022',
  
  description: 'Hello! Let`s start cleaning!',
  estimation: '5',
  failedPomodoros: [],
  priority: 'middle',
  status: 'GLOBAL_LIST',
  title: 'Test',
};
