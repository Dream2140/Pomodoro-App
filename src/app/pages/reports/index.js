import { ReportsController } from './scripts/reportsController';
import { ReportsModel } from './scripts/reportsModel';
import { ReportsView } from './scripts/reportsView';

import './styles/reports.less';

const reportsModel = new ReportsModel();
const reportsView = new ReportsView(reportsModel);

/* eslint-disable no-unused-vars */
const reportsController = new ReportsController(reportsView, reportsModel);
/* eslint-disable no-unused-vars */