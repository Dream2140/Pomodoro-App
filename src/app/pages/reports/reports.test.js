/* eslint-disable no-undef */
import reports from './template/reports.hbs';
import { ReportsModel } from './scripts/reportsModel';
import { tasks } from './__mocks__/tasksReport';
import { tabsData } from './__mocks__/tabsData';
import { dataForPomodorosReport } from './__mocks__/dataForPomodorosReport';
import { dataForTasksReport } from './__mocks__/dataForTasksReport';

jest.mock('./scripts/reportsView.js');
jest.mock('../../helpers/settingsData');

const reportsModel = new ReportsModel();

describe('check report rendering', () => {
  it('renders correctly report', () => {
    const tree = JSON.stringify(reports(tabsData));
    expect(tree).toMatchSnapshot();
  });
});

describe('createReport', () => {
  beforeEach(() => {
    reportsModel.tasks = tasks;
    jest.spyOn(reportsModel, 'sortReportsByPeriod');
    jest.spyOn(reportsModel, 'createReportsData');
  });

  it('shold start creating reports', () => {
    reportsModel.createReport('tasks', 1, 'day');
    expect(reportsModel.sortReportsByPeriod).toBeCalled();
    expect(reportsModel.createReportsData).toBeCalled();
  });

  it('shold not start creating reports', () => {
    reportsModel.createReport('tasks');
    expect(reportsModel.sortReportsByPeriod).not.toBeCalled();
    expect(reportsModel.createReportsData).not.toBeCalled();
  });
});

describe('should sort tasks', () => {
  it('shold sort reports for 1 day', () => {
    const result = reportsModel.sortReportsByPeriod(tasks, 1);
    expect(result).toEqual(tasks);
  });

  it('shold sort reports for 7 weeks', () => {
    const result = reportsModel.sortReportsByPeriod(tasks, 7);
    expect(result).toEqual(tasks);
  });

  it('shold return falsy result', () => {
    const result = reportsModel.sortReportsByPeriod(true, 7);
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.sortReportsByPeriod(true, 0);
    expect(result).toBeFalsy();
  });
});

describe('should create data for pomodoros', () => {
  it('shold return data for pomodoros report', () => {
    const result = reportsModel.createDataPomodoros(tasks);
    expect(result).toEqual(dataForPomodorosReport);
  });

  it('shold return falsy result', () => {
    const result = reportsModel.createDataPomodoros();
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.createDataPomodoros([]);
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.createDataPomodoros(67);
    expect(result).toBeFalsy();
  });
});

describe('should create data for tasks report', () => {
  it('shold return data for tasks report', () => {
    const result = reportsModel.createDataTasks(tasks);
    expect(result).toEqual(dataForTasksReport);
  });

  it('shold return falsy result', () => {
    const result = reportsModel.createDataTasks([]);
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.createDataTasks();
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.createDataTasks(6);
    expect(result).toBeFalsy();
  });
});

describe('should check if task successfull', () => {
  it('shold return truthy result', () => {
    const result = reportsModel.isSuccessfullTask(tasks[1]);
    expect(result).toBeTruthy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.isSuccessfullTask(tasks[0]);
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.isSuccessfullTask();
    expect(result).toBeFalsy();
  });

  it('shold return falsy result', () => {
    const result = reportsModel.isSuccessfullTask(6);
    expect(result).toBeFalsy();
  });
});
