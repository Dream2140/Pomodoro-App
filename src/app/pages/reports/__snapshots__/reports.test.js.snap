// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check report rendering renders correctly report 1`] = `"\\"<header class=\\\\\\"header \\\\\\">\\\\r\\\\n    <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n        <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                    <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                </a>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                            data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                </ul>\\\\r\\\\n            </nav>\\\\r\\\\n\\\\r\\\\n        </div>\\\\r\\\\n\\\\r\\\\n    </div>\\\\r\\\\n</header><main class=\\\\\\"content reports-page\\\\\\">\\\\r\\\\n\\\\r\\\\n       <section class=\\\\\\"reports\\\\\\">\\\\r\\\\n        <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n            <div class=\\\\\\"reports__container\\\\\\">\\\\r\\\\n                <div class=\\\\\\"tab \\\\\\">\\\\r\\\\n                    <div class=\\\\\\"tab__container\\\\\\">\\\\r\\\\n                        <ul class=\\\\\\"tab__item \\\\\\">\\\\r\\\\n                        </ul>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>                <div class=\\\\\\"reports__content\\\\\\">\\\\r\\\\n                    <a href=\\\\\\"/task-list\\\\\\" class=\\\\\\"reports__link-left icon icon-arrow-left\\\\\\" title=\\\\\\"Go to Global List\\\\\\"></a>   \\\\r\\\\n                    <div class=\\\\\\"reports__graph\\\\\\" id=\\\\\\"reportsGraph\\\\\\">\\\\r\\\\n                        \\\\r\\\\n                    </div>\\\\r\\\\n                </div>\\\\r\\\\n                <div class=\\\\\\"tab \\\\\\">\\\\r\\\\n                    <div class=\\\\\\"tab__container\\\\\\">\\\\r\\\\n                        <ul class=\\\\\\"tab__item \\\\\\">\\\\r\\\\n                        </ul>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>            </div>\\\\r\\\\n        </div>\\\\r\\\\n       </section>\\\\r\\\\n\\\\r\\\\n\\\\r\\\\n    </main>\\""`;
